src/main/webapp/app/modeling-assessment-editor/modeling-assessment-editor.component.ts

```diff
@@ -372,12 +385,50 @@ export class ModelingAssessmentEditorComponent implements OnInit, OnDestroy {
         }
     }
 
+    /**
+     * Add all elements for which no corresponding feedback element exist to the map of highlighted elements. To make sure that we do not have outdated elements in the map, all
+     * elements with the corresponding "missing feedback color" get removed first.
+     */
     private highlightElementsWithMissingFeedback() {
-        this.highlightedElementIds = new Set<string>();
-        this.model!.elements.forEach((element: UMLElement) => {
-            if (this.referencedFeedback.findIndex(feedback => feedback.referenceId === element.id) < 0) {
-                this.highlightedElementIds.add(element.id);
+        if (!this.model) {
+            return;
+        }
+
+        this.highlightedElements = this.highlightedElements
```

Comment:
Reading this again today, I can only emphasize the negative impact on the readability from the use of `Maps` here.

E.g.:
` this.highlightedElements = this.highlightedElements
            ? this.removeHighlightedFeedbackOfColor(this.highlightedElements, FeedbackHighlightColor.RED)
            : new Map<string, string>();`
vs.
`highlightedElements: Array<[string, string]> = [];`
...
`this.highlightedElements = this.highlightedElements.filter(([elem, color]) => color !== FeedbackHighlightColor.RED)`

If you don't want to make highlightedElements an assoc array, you could make it an object and use lodash fp:
`import {fromPairs, filter, toPairs} from 'lodash/fp';`
`this.highlightedElements = compose(fromPairs, filter(([elem, color]) => color !== FeedbackHighlightColor.RED)), toPairs)(this.highlightedElements);`

Alternatively in plain JS:

`this.highlightedElements = Object.entries(this.highlightedElements).filter(...).reduce((acc, [key, val]) => ({[key]: val, ...acc}), {});`

Reply:
To clarify: this is an optional improvement, the code works.

Reply:
As discussed, we will think about it in a follow-up PR.